packages/svelte/src/internal/client/dom/template.js

```diff
@@ -236,10 +208,18 @@ export function comment() {
  * Assign the created (or in hydration mode, traversed) dom elements to the current block
  * and insert the elements into the dom (in client mode).
  * @param {Text | Comment | Element} anchor
- * @param {import('#client').Dom} dom
+ * @param {DocumentFragment | Element | Comment} node
  */
-export function append(anchor, dom) {
+export function append(anchor, node) {
 	if (!hydrating) {
-		anchor.before(/** @type {Node} */ (dom));
+		/** @type {import('#client').Dom} */
+		const dom =
+			node.nodeType === 11
+				? /** @type {import('#client').TemplateNode[]} */ ([...node.childNodes])
```

Comment:
This seems to cause a memory leak. We shouldn't reference all child nodes, as those nodes might get removed from internal effects – yet still get referenced here and thus retained. I should have noticed this on my review, sigh.